newlib/libc/machine/riscv/memcpy-asm.S

```diff
@@ -28,5 +28,91 @@ memcpy:
 2:
   ret
 
-  .size	memcpy, .-memcpy
+#else
+.Lwordify:
+    andi   a3, a0, 3
+    beqz   a3, .Laligned
```

Comment:
as an additional improvement for performance, you can explore if going into a different alignment function based on the value of a3 could be an option. I.e. have alignment functions that use lw and or lh to achive quicker alignment.
The alternative is to use a similar approach as in memset, where you jump into a sequence of sb/lb instructions at an offset. For sure this would be better than the current solution, but I would like to also explore what i mentioned above